src/p_spec.c

```diff
@@ -2309,7 +2323,8 @@ void P_UpdateSpecials (void)
           flattranslation[anim->basepic + i] = pic;
           // [crispy] add support for SMMU swirling flats
           if (anim->speed > 65535 || anim->numpics == 1 || r_swirl)
-            flattranslation[anim->basepic + i] = -1;
+            if (!Swirl_CustomRule(anim->basepic, swirl_custom_rule))
+              flattranslation[anim->basepic + i] = -1;
```

Comment:
Wouldn't this approach sabotage all PWAD authors who took the effort to define individual swirling flats in their bundled ANIMATD lump?